specification/compute/Compute.Management/VirtualMachineScaleSetExtension.tsp

```diff
@@ -0,0 +1,94 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+import "./VirtualMachineScaleSet.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Compute;
+/**
+ * Describes a Virtual Machine Scale Set Extension.
+ */
+@parentResource(VirtualMachineScaleSet)
+model VirtualMachineScaleSetExtension
+  is Azure.ResourceManager.ProxyResource<VirtualMachineScaleSetExtensionProperties> {
+  ...ResourceNameParameter<
```

Comment:
Name of the extension should be settable per discussion. Did we solve this?

Reply:
I saw that this should be solved already, at least in Swagger. Though why remove this solution in this commit?
https://github.com/Azure/azure-rest-api-specs/pull/32748/commits/9944fe9c72137d7e7e05223de03df0d3d7f16a29#diff-90480f5f9ca88a7d93e2a523edb5dfacaf03e45bfb4de4eb4c57f837867abfc3

Java could make above java-only, just curious why.